File: flutter_ui/lib/core/services/translateService.dart
import 'dart:collection';
import 'dart:convert';

import 'package:flutter_ui/core/interceptors/authInterceptor.dart';
import 'package:flutter_ui/core/models/deleteModel.dart';
import 'package:flutter_ui/core/models/response/listResponseModel.dart';
import 'package:flutter_ui/core/models/response/responseModel.dart';
import 'package:flutter_ui/core/models/response/singleResponseModel.dart';
import 'package:flutter_ui/core/models/translate/translateAddModel.dart';
import 'package:flutter_ui/core/models/translate/translateModel.dart';
import 'package:flutter_ui/core/utilities/service.dart';
import 'package:flutter_ui/core/utilities/serviceRepository.dart';
import 'package:flutter_ui/environments/api.dart';

class TranslateService extends Service
    with ServiceRepository<TranslateAddModel, TranslateModel> {
  @override
  Future<ResponseModel> add(TranslateAddModel addModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "translates/add"),
      body: addModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ResponseModel> delete(DeleteModel deleteModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "translates/delete"),
      body: deleteModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ListResponseModel<TranslateModel>> getAll() async {
    var response = await httpClient.get(
      Uri.parse(API_URL + "translates/getall"),
    );

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  @override
  Future<SingleResponseModel<TranslateModel>> getById(int id) async {
    var response = await httpClient.get(
      Uri.parse(
        API_URL + "translates/getbyid?id=" + id.toString(),
      ),
    );

    return SingleResponseModel<TranslateModel>.fromJson(response);
  }

  @override
  Future<ResponseModel> update(TranslateModel updateModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "translates/add"),
      body: updateModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  Future<ListResponseModel<TranslateModel>> getByKey(String key) async {
    var response = await httpClient.get(
      Uri.parse(API_URL + "translates/getbykey?key=" + key),
    );

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  Future<ListResponseModel<TranslateModel>> getByLanguageId(
      int languageId) async {
    var response = await httpClient.get(
      Uri.parse(API_URL +
          "translates/getbylanguageid?languageId=" +
          languageId.toString()),
    );

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  Future<Map> getTranslates(String languageCode) async {
    var response = await httpClient.get(
      Uri.parse(
          API_URL + "translates/gettranslates?languageCode=" + languageCode),
    );

    return Map.from(json.decode(response.body));
  }
}

File: AngularUI/angular-ui/src/app/core/services/translate.service.ts
import { TranslateModel } from './../models/translate/translateModel';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ListResponseModel } from '../models/response/listResponseModel';
import { ResponseModel } from '../models/response/responseModel';
import { SingleResponseModel } from '../models/response/singleResponseModel';
import { apiUrl } from 'src/api';
import { ServiceRepository } from './service-repository';
import { DeleteModel } from '../models/deleteModel';

@Injectable({
  providedIn: 'root',
})
export class TranslateService
  implements ServiceRepository<TranslateModel, number>
{
  constructor(private httpClient: HttpClient) {}

  add(addModel: TranslateModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/add';
    return this.httpClient.post<ResponseModel>(newPath, addModel);
  }

  delete(deleteModel: DeleteModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/delete';
    return this.httpClient.post<ResponseModel>(newPath, deleteModel);
  }

  update(updateModel: TranslateModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/update';
    return this.httpClient.post<ResponseModel>(newPath, updateModel);
  }

  getById(id: number): Observable<SingleResponseModel<TranslateModel>> {
    let newPath = apiUrl + 'translates/getbyid?id=' + id;
    return this.httpClient.get<SingleResponseModel<TranslateModel>>(newPath);
  }

  getAll(): Observable<ListResponseModel<TranslateModel>> {
    let newPath = apiUrl + 'translates/getall';
    return this.httpClient.get<ListResponseModel<TranslateModel>>(newPath);
  }

  getByKey(key: string): Observable<ListResponseModel<TranslateModel>> {
    let newPath = apiUrl + 'translates/getbykey?key=' + key;
    return this.httpClient.get<ListResponseModel<TranslateModel>>(newPath);
  }

  getByLanguageId(
    languageId: number
  ): Observable<ListResponseModel<TranslateModel>> {
    let newPath =
      apiUrl + 'translates/getbylanguageid?languageId=' + languageId;
    return this.httpClient.get<ListResponseModel<TranslateModel>>(newPath);
  }

  getTranslates(languageCode: string): Observable<SingleResponseModel<any>> {
    let newPath =
      apiUrl + 'translates/gettranslates?languageCode=' + languageCode;
    return this.httpClient.get<SingleResponseModel<any>>(newPath);
  }
}
